packages/flutter/lib/src/rendering/mouse_cursor.dart

```diff
@@ -0,0 +1,270 @@
+/// that represents the same concept, such as text, being clickable, or forbidden
+/// operation. Since the set of system cursors supported by each platform varies,
+/// multiple instances can correspond to the same system cursor.
+///
+/// [SystemMouseCursors] enumerates the complete set of system cursors supported
+/// by Flutter, which are hard-coded in the platform engine. Therefore, manually
+/// instantiating this class is neither useful nor supported.
+class SystemMouseCursor extends PreparedMouseCursor {
+  // Application code shouldn't directly instantiate system mouse cursors, since
+  // the supported system cursors are enumerated in [SystemMouseCursors].
+  const SystemMouseCursor._({
+    @required this.shapeCode,
+    @required this.debugDescription,
+  }) : assert(shapeCode != null),
+       assert(debugDescription != null);
+
+  /// A globally unique number that identifies the shape of the cursor.
+  ///
+  /// The exact value of a [shapeCode] is intentionally random, and its
+  /// interpretation (i.e. corresponding system cursor) is hard-coded in the
+  /// platform engine.
+  ///
+  /// See the documentation of [SystemMouseCursor] for introduction.
+  final int shapeCode;
+
+  @override
+  final String debugDescription;
+
+  @override
+  bool operator ==(dynamic other) {
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is SystemMouseCursor
+        && other.shapeCode == shapeCode;
+  }
+
+  @override
+  int get hashCode => shapeCode;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(IntProperty('shapeCode', shapeCode, level: DiagnosticLevel.debug));
+  }
+}
+
+/// A collection of system [MouseCursor]s.
+///
+/// System cursors are mouse cursors provided by the current platform, available
+/// without external resources, and are therefore the most common cursors.
+///
+/// [SystemMouseCursors] is a superset of the system cursors of every platform
+/// that Flutter supports, therefore some of these objects might map to the same
+/// result, or fallback to the basic arrow. This mapping is implemented by the
+/// Flutter engine that the program is running on.
+/// 
+/// The cursor names are chosen to reflect the cursors' use cases instead of
+/// their shapes, because different platforms might (although not common) use
+/// different shapes for the same use case.
+class SystemMouseCursors {
+  // This class only contains static members, and should not be initiated or
+  // extended.
+  factory SystemMouseCursors._() => null;
+
+  /// A special value that tells Flutter to release the control of cursors.
+  ///
+  /// A region with this value will absorb the search for mouse cursor
+  /// configuration, but the pointer's cursor will not be changed when it enters
+  /// or is hovering this region. In other words, Flutter has released its
```

Comment:
Just eliminate the "In other words" sentence.  If there is a simpler way to describe it, just use that way instead of saying it twice.